File: app/src/main/java/com/fittrack/data/entity/RoutineExercise.kt
package com.fittrack.data.entity

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(
    tableName = "routine_exercises",
    foreignKeys = [
        ForeignKey(
            entity = WorkoutRoutine::class,
            parentColumns = ["id"],
            childColumns = ["routineId"],
            onDelete = ForeignKey.CASCADE
        ),
        ForeignKey(
            entity = Exercise::class,
            parentColumns = ["id"],
            childColumns = ["exerciseId"],
            onDelete = ForeignKey.CASCADE
        )
    ],
    indices = [Index("routineId"), Index("exerciseId")]
)
data class RoutineExercise(
    @PrimaryKey(autoGenerate = true) val id: Int = 0,
    val routineId: Int,
    val exerciseId: Int,
    val orderIndex: Int,
    val defaultSets: Int = 3,
    val defaultReps: Int = 10
)

File: ios/P3Fitness/Models/RoutineExercise.swift
import SwiftData
import Foundation

@Model
final class RoutineExercise {
    var id: Int
    var routineId: Int
    var exerciseId: Int
    var exerciseName: String
    var orderIndex: Int
    var defaultSets: Int
    var defaultReps: Int

    init(routineId: Int, exerciseId: Int, exerciseName: String, orderIndex: Int, defaultSets: Int = 3, defaultReps: Int = 10) {
        self.id = Int(Date().timeIntervalSince1970 * 1000) % 1_000_000_000
        self.routineId = routineId
        self.exerciseId = exerciseId
        self.exerciseName = exerciseName
        self.orderIndex = orderIndex
        self.defaultSets = defaultSets
        self.defaultReps = defaultReps
    }
}
